### PR TITLE
x-pack/filebeat/module/o365: fix handling of o365.audit.OperationProperties

### LINKED ISSUE #37800: [o365] Could not index event to Elasticsearch
Filebeat 8.9.1 is fetching logs from Office365/Microsoft365 management activity API and sending them to Elasticsearch via Logstash. Recetly I observe raise in `Could not index event to Elasticsearch` errors:

> {"create"=>{"_index"=>".ds-logs-filebeat.8.9.2-o365-2024.01.23-000004", "_id"=>"-NreWo0BSez6H-H_YS06", "status"=>400, "error"=>{"type"=>"document_parsing_exception", "reason"=>"[1:1128] object mapping for [o365.audit.OperationProperties] tried to parse field [OperationProperties] as object, but found a concrete value"}}

Sample event:

```
{
  "event": {
    "action": "CreateArtifact",
    "category": "web",
    "dataset": "o365.audit",
    "code": "PowerBIAudit",
    "module": "o365",
    "kind": "event",
    "outcome": "success",
    "id": "a4420e70-b7a1-4d5b-8175-11e3364acd22",
    "provider": "PowerBI",
    "type": "info"
  },
  "o365": {
    "audit": {
      "Activity": "CreateArtifact",
      "WorkspaceName": "obszar_robaczy",
      "OrganizationId": "53d83e1d-xxx-xxx-84e9-01ec5045dd81",
      "Operation": "CreateArtifact",
      "Id": "a4420e70-b7a1-xxx-xxx-11e3364acd22",
      "CreationTime": "2024-01-30T14:23:40",
      "Timestamp": "2024-01-30T14:22:50",
      "UserId": "username@domain.pl",
      "ClientIP": "31.135.18.197",
      "RecordType": 20,
      "ResultStatus": "InProgress",
      "ObjectDisplayName": "test_lakehouse",
      "OperationId": "a84f7f73-xxxx-xxxx-8cf3-094f69c23756",
      "Experience": "Lakehouse",
      "WorkspaceId": "91dad513-xxxx-xxxx-94bb-f5cbf305691c",
      "ObjectId": "0e00d1cf-825a-4d78-98ff-8a8199357669",
      "UserAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36",
      "Workload": "PowerBI",
      "RequestId": "fcbbe282-xxx-xxxx-xxxx-dc1e6d9b090b",
      "OperationProperties": "[{\"Name\":\"SystemArtifactType\",​\"Value\":\"None\"}]",
      "ObjectType": "Lakehouse",
      "UserType": 0,
      "UserKey": "xxxxxxxx"
    }
  },
  "@version": "1",
  "input": {
    "type": "o365audit"
  },
  "ABS_SIEM_OUTPUT": {
    "index_name": "logs-filebeat.8.9.2-o365",
    "pipeline": "filebeat-8.9.2-o365-audit-pipeline"
  },
  "source": {
    "ip": "31.135.11.111"
  },
  "network": {
    "type": "ipv4"
  },
  "host": {
    "id": "53d83e1d-82ae-4273-84e9-01ec5045dd81",
    "ip": "31.135.11.111",
    "name": "assecobusinesssolutions.onmicrosoft.com"
  },
  "related": {
    "user": "username",
    "ip": "31.135.11.111"
  },
  "@timestamp": "2024-01-30T14:23:40.000Z",
  "service": {
    "type": "o365"
  },
  "client": {
    "ip": "31.135.11.111",
    "address": "31.135.11.111"
  },
  "organization": {
    "id": "53d83e1d-1111-111-84e9-01ec5045dd81",
    "name": "53d83e1d-1111-111-84e9-01ec5045dd81"
  },
  "user": {
    "email": "username@domain.pl",
    "id": "username@domain.pl",
    "domain": "domain.pl",
    "name": "username"
  },
  "user_agent": {
    "original": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
  },
  "ecs": {
    "version": "1.12.0"
  },
  "data_stream": {
    "dataset": "filebeat.8.9.2",
    "type": "logs",
    "namespace": "o365"
  },
  "agent": {
    "version": "8.9.2",
    "ephemeral_id": "567e8840-1111-1111-1111-671d671de964",
    "id": "02258696-1111-1111-1111-85404ad17474",
    "name": "filebeat-downloader-1",
    "type": "filebeat"
  },
  "tags": [
    "forwarded",
    "beats",
    "logstash_pipeline_beats",
    "beats_input_raw_event"
  ],
  "fileset": {
    "name": "audit"
  }
}
```

In other events OperationProperties is list of objects with "Name" and "Value" keys.

### LINKED ISSUE #38278: [filebeat][o365] Mapping problem on o365.audit.AdditionalInfo
Hi,

I've observed that the O365 module in FileBeat sometimes treats o365.audit.AdditionalInfo as a JSON string, and sometimes as an object with some sub-fields instead.  Why do I say this?  In Kibana, it will report this field as having a type conflict, and when I look at the text version of this field it's a JSON string.  The non-text version of this field has sub-fields.  I've seen this happen with FileBeat version 7.17(.7?) and with version 8.12.0.

I did post a note about this on the [Elastic discussion forum](https://discuss.elastic.co/t/filebeat-o365-module-field-type-inconsistency/352884), but never received any response about it.

I'm running Filebeat on Ubuntu 22.04.  I don't push the O365 logs directly to Elasticsearch -- I push them to Logstash using a Beats input, and from there push to Elasticsearch.  I've worked around this issue by having Logstash check to see if this field is a string, and convert it to an object if it is a JSON string,

Thanks,

      - Daniel

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

In some cases, this field may be a string. This results in a mapping failure. So in cases where the field is a string, conditionally parse out the JSON.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Closes #37800
- Closes #38278

### CODE REVIEW

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼

**Comment by efd6** [comment]
Waiting on input from reporter for answer to question ([here](https://github.com/elastic/beats/issues/37800#issuecomment-2033372193) and [here](https://github.com/elastic/beats/issues/38278#issuecomment-2033426720)) before merging.

**Pull request merged**
